src/chains/ethereum/console.log/console.sol

```diff
@@ -0,0 +1,2799 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >= 0.4.22 <0.9.0;
+
+library console {
+    address constant CONSOLE_ADDRESS = address(0x000000000000000000636F6e736F6c652e6c6f67);
+
+    function _sendLogPayload(bytes memory payload) private view {
+        address consoleAddress = CONSOLE_ADDRESS;
+        assembly {
+            let argumentsLength := mload(payload)
+            let argumentsOffset := add(payload, 32)
+            pop(staticcall(gas(), consoleAddress, argumentsOffset, argumentsLength, 0, 0))
```

Comment:
The result of the call to the staticcall must be handled somehow. The `pop` removes the result from the stack. An alternative is to assign it to a variable:

```solidity
let trash := staticcall(gas(), consoleAddress, argumentsOffset, argumentsLength, 0, 0)
```


Which basically does the same thing (it might even end up writing the same op codes, I haven't checked) and cost the same amount of gas either way.

Reply:
Neat! Might be worth a comment :shrug:

Reply:
Maaybbeee... it just that this is: 

 * assembly in a stack machine VM (last in first out)
 * and I already find `mload` and `add` to be way more confusing :sweat_smile: